### DISCUSSION TITLE
Include sequence as method vs a custom TextProperty

### DISCUSSION THREAD

**Discussion by optmsp** [spatie/icalendar-generator#88] [Ideas]
Unless I'm missing something, to set the SEQUENCE in an event, you have to do this as a custom TextProperty to your Event, like so:

```appendProperty(TextProperty::create('SEQUENCE', $seqID))```

Since the SEQUENCE property is essential when sending any changes to a client, e.g., if sending an ICS in an email to modify or cancel an event, this seems like it should be built right into the library.

My thinking is that we would have these options:

* No sequence() method used. Does not include SEQUENCE.
* sequence() used. Uses time() as the SEQUENCE.
* sequence(int $seq) used. Uses whatever is supplied by the caller.

I'm happy to do a PR on this if this is accepted as worth-while need.

**Comment by freekmurze** [comment]
Feel free to create a PR. Make sure to update the docs and tests.

**Comment by theriftlab** [comment]
@optmsp did you get anywhere with this? The only issue I see is allowing `sequence()` with no value & defaulting to `time()` - this could be potentially disastrous for setting up a feed since the same event will have a different SEQUENCE each time it is polled. I would assume only the underlying data itself will/should ever know which iteration it is publishing.
